src/AdminConnection.ts

```diff
@@ -1,1213 +1,1218 @@
-                    if (data === PERMISSION_ERROR) {
-                        reject('May not read "getCompactRepository"');
-                    } else if (!data) {
-                        reject('Cannot read "getCompactRepository"');
-                    } else {
-                        resolve(data);
-                    }
-                }
-            });
-        });
-
-        return this._promises.repoCompact;
-    }
-
-    /**
-     * Get the list of all hosts in compact form (only _id, common.name, common.icon, common.color, native.hardware.networkInterfaces)
-     * @param {boolean} [update] Force update.
-     * @returns {Promise<ioBroker.Object[]>}
-     */
-    getCompactHosts(update: boolean): Promise<ioBroker.Object[]> {
-        if (Connection.isWeb()) {
-            return Promise.reject('Allowed only in admin');
-        }
-        if (!update && this._promises.hostsCompact) {
-            return this._promises.hostsCompact;
-        }
-
-        if (!this.connected) {
-            return Promise.reject(NOT_CONNECTED);
-        }
-
-        this._promises.hostsCompact = new Promise((resolve, reject) =>
-            this._socket.emit('getCompactHosts', (err, systemConfig) =>
-                err ? reject(err) : resolve(systemConfig)));
-
-        return this._promises.hostsCompact;
-    }
-}
+import { Connection, ERRORS } from "./Connection";
+import type { ConnectionProps } from "./ConnectionProps";
+import type {
+	AdminEmitEvents,
+	AdminListenEvents,
+	CompactAdapterInfo,
+	CompactHost,
+	CompactInstalledInfo,
+	CompactInstanceInfo,
+	CompactRepository,
+	Logfile,
+} from "./SocketEvents";
+import {
+	getObjectViewResultToArray,
+	normalizeHostId,
+	objectIdToHostname,
+} from "./tools";
+
+interface Certificate {
+	name: string;
+	type: "public" | "private" | "chained";
+}
+
+function parseCertificate(name: string, cert: string): Certificate | undefined {
+	if (!cert) return;
+
+	let type: Certificate["type"];
+	// If it is filename, it could be everything
+	if (
+		cert.length < 700 &&
+		(cert.indexOf("/") !== -1 || cert.indexOf("\\") !== -1)
+	) {
+		if (name.toLowerCase().includes("private")) {
+			type = "private";
+		} else if (cert.toLowerCase().includes("private")) {
+			type = "private";
+		} else if (name.toLowerCase().includes("public")) {
+			type = "public";
+		} else if (cert.toLowerCase().includes("public")) {
+			type = "public";
+		} else {
+			// TODO: is this correct?
+			return undefined;
+		}
+	} else {
+		type =
+			cert.substring(0, "-----BEGIN RSA PRIVATE KEY".length) ===
+				"-----BEGIN RSA PRIVATE KEY" ||
+			cert.substring(0, "-----BEGIN PRIVATE KEY".length) ===
+				"-----BEGIN PRIVATE KEY"
+				? "private"
+				: "public";
+
+		if (type === "public") {
+			const m = cert.split("-----END CERTIFICATE-----");
+			if (
+				m.filter((t) => t.replace(/\r\n|\r|\n/, "").trim()).length > 1
+			) {
+				type = "chained";
+			}
+		}
+	}
+	return { name, type };
+}
+
+interface IPAddress {
```

Comment:
Why are this interface and the next not exported?

Reply:
The next is only used internally - I'm not even sure why it was used that way. This one should be exported, I agree.